where-to-travel/src/main/webapp/script.js

```diff
@@ -1,141 +1,302 @@
+
+    placeMarker.addListener('click', function () {
+      toggleFocusOff();
+      focussedPin = placeMarker;
+      selectLocationCard(placeMarker.getTitle());
+      placeMarker.setIcon('icons/selectedPin.svg');
     });
 
-    placeMarker.addListener('click', function() {
-      infowindow.open(map, placeMarker);
+    placeMarker.addListener('mouseover', function () {
+      placeMarker.setIcon('icons/selectedPin.svg');
+    });
+
+    placeMarker.addListener('mouseout', function () {
+      if (placeMarker != focussedPin) {
+        placeMarker.setIcon('icons/pin.svg');
+      }
     });
+
+    markers.push(placeMarker);
   }
+  $('.icon').click(function() {
+    $(this).toggleClass('press');
+  });
+}
+
+/**
+ * Helper function that returns the an HTML string representing a place card
+ * that can be added to the DOM.
+ * @param {string} title the place title
+ * @param {string} directionsLink the link to the GMaps directions for this place
+ * @param {string} timeStr the amount of time it takes to travel to this place, as a string
+ */
+function getLocationCardHtml(title, directionsLink, timeStr) {
+  const iconId = 'icon' + title;
+  return innerHtml = '' +
+    `<div class="card location-card" placeName="${title}">
+      <div class="card-body">
+        <h5 class="card-title">${title}
+        <span class="icon" id="${iconId}">
+          &#9733
+        <span>
+        </h5>
+        <a target="_blank" href="${directionsLink}" class="badge badge-primary">Directions</a>
+        <p>${timeStr}</p>
+        <i></i>
+      </div>
+    </div>`;
+}
+
+function getLoginHtml() {
+  return `<a class="btn btn-outline-primary" style="text-align: center" href="login.html">Login</a>
+          <span id="nav-text">or</span>
+          <a class="btn btn-outline-primary" href="signup.html">Sign up</a>`;
+}
+
+function getUserDashHtml(user) {
+  return '<a class="btn btn-outline-primary" style="text-align: center" href="login.html">Logout</a>';
 }
- 
 
-/** 
- * If browser supports geolocation and user provides permissions, obtains user's 
+/**
+ * Given a title, selects the corresponding marker by focussing it
+ * @param {string} title the name of the place whose marker to focus
+*/
+function selectLocationMarker(title) {
+  for (marker of markers) {
+    if (marker.getTitle() == title) {
+      focussedPin = marker;
+      marker.setIcon('icons/selectedPin.svg');
+    }
+  }
+}
+
+/**
+ * Given a title, selects the corresponding card by focussing it
+ * @param {string} title the name of the place whose card to focus
+ */
+function selectLocationCard(title) {
+  scrollWindow = document.getElementById(scrollId);
+  for (locationCard of scrollWindow.childNodes) {
+    if (locationCard.hasChildNodes() && locationCard.getAttribute("placeName") == title) {
+      locationCard.classList.add("active");
+      focussedCard = locationCard;
+    }
+  }
+}
+
+/** Clears all place cards that are currently displayed. */
```

Comment:
update comment to also say markers are cleared

Reply:
fixed in 06f03d0